### ISSUE TITLE
OneSignalNotificationServiceExtension bitcode error on build

### ISSUE THREAD

**Issue by Style87** [OneSignal/OneSignal-iOS-SDK#659]
**Description:**

When I go to build to my target device I get the error below.

ld: '/Users/lucashartzell/Library/Developer/Xcode/DerivedData/Runner-dpeadbdaauvfpbgyhkflubgrwpbm/Build/Products/Debug-iphoneos/libPods-OneSignalNotificationServiceExtension.a(Pods-OneSignalNotificationServiceExtension-dummy.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. file '/Users/lucashartzell/Library/Developer/Xcode/DerivedData/Runner-dpeadbdaauvfpbgyhkflubgrwpbm/Build/Products/Debug-iphoneos/libPods-OneSignalNotificationServiceExtension.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I have bitcode_enabled set to no everywhere it can be. I've looked through all the values in Xcode and I've greped for it in the code base to confirm so.

**Environment**
iOS 13.4                      	-sdk iphoneos13.4
The one signal sdk was added through cocoapods. The onesignal_flutter version in the pubspec is 2.3.4.

**Steps to Reproduce Issue:**

1. Install iOS SDK version 2.8.2 with Cocoapods into project
2. Attempt to build to a physical device

**Anything else:**
I have already tried running flutter clean and doing a full reinstall on the pods with no luck.

[Podfile.txt](https://github.com/OneSignal/OneSignal-iOS-SDK/files/4568650/Podfile.txt)

**Comment by Graig123git** [comment]
@Style87 +1, did you manage to figure this out ?

**Comment by Style87** [comment]
@Graig123git, unfortunately I have not. I've been at this error off and on for a weeks now.

**Comment by rgomezp** [comment]
It looks like bitcode is still enabled somewhere. The problem arises since the app was compiled without bitcode. Make sure it is set to No in _both_ the project and library settings

**Comment by Style87** [comment]
Is it possible that grepping my entire code base from the root for BITCODE would not show an instance where it's set to yes? I have done this and every value is set to no.

**Comment by osmanc777** [comment]
How about you guys, this error is really a headache since a few days ago everything was working fine for me but suddenly I threw myself that error when trying to subscribe the user.

Today I managed to find a solution or at least make it work.

Steps to follow:

1. Have the Push Notifications added in Xcode
enable these options by the background push:
-Background Fetch
-Remote notifications

![xcodepush](https://user-images.githubusercontent.com/23218509/83215263-c6d50d00-a123-11ea-94a8-400d49a511bd.png)


2. Check if they have any updates to cocoa pods
```sh
pod repo update
```
![pods1](https://user-images.githubusercontent.com/23218509/83217643-7660ae00-a129-11ea-8386-0a1871b8c3ce.png)

```
sudo gem install cocoapods
```
![pods2](https://user-images.githubusercontent.com/23218509/83217698-94c6a980-a129-11ea-9aaa-9c32823e31c6.png)

then in the root folder of your project run again
```
pod repo update
```
![pods3](https://user-images.githubusercontent.com/23218509/83217765-b6279580-a129-11ea-9967-7431e6d4bf5e.png)

in my case as I am working with ionic, enable onesignal debugging options, building an object of each persimo that requests onesignal

```ts
      this.oneSignal.startInit('#####', ####');

      //important debuggin info
      this.oneSignal.setLogLevel({logLevel: 6, visualLevel: 0});

      this.oneSignal.inFocusDisplaying(this.oneSignal.OSInFocusDisplayOption.Notification);

      this.oneSignal.handleNotificationReceived().subscribe((respNoti) => {
        // do something when notification is received
      });

      this.oneSignal.handleNotificationOpened().subscribe((respNoti) => {
        // do something when a notification is opened
      });

      this.oneSignal.getIds().then(info => {
        this.userId = info.userId;
        this.userToken = info.pushToken;
        console.log('ID OneSignal: ',this.userId);
        console.log('Token push: ', this.userToken);
      });

      this.oneSignal.endInit();
```
//function get log status user (debug for safari)
```ts
  getStatusUser() {
    this.oneSignal.getPermissionSubscriptionState().then((resp:OSPermissionSubscriptionState) => {
      let Obj:any = {
        Promt:       resp.permissionStatus.hasPrompted,
        PermisosIOS: resp.permissionStatus.status,
        PermisosAnd: resp.permissionStatus.state,
        Suscrito:    resp.subscriptionStatus.subscribed,
        ConfigUser:  resp.subscriptionStatus.userSubscriptionSetting,
        UserID:      resp.subscriptionStatus.userId,
        UserToken:   resp.subscriptionStatus.pushToken
      }


      console.log('OneDebug: ',Obj);
      console.log('iOS: 0 = Not Determined, 1 = Denied, 2 = Authorized ', resp.permissionStatus.status);
      console.log('Android: 1 = Authorized, 2 = Denied ', resp.permissionStatus.state);
      console.log('FCM/APNs: ', resp.subscriptionStatus.pushToken);
    })
  }
```
![pushInfo](https://user-images.githubusercontent.com/23218509/83220030-2684e580-a12f-11ea-83d9-87927e6b5e77.png)



then proceed to generate project build again

to know if your push subscription was successful the userId and pushToken must contain a string of strings must not be null

basically the problem is fixed when updating cocoa pods.

**Comment by rgomezp** [comment]
Thank you for the solution! I will close this issue. Cheers

**Comment by nomicool** [comment]
The solution did not help

**Comment by robertohuertasm** [comment]
I'm getting this in Flutter too. @rgomezp should we reopen this issue?

**Comment by rgomezp** [comment]
Howdy @nomicool @robertohuertasm 

Before reopening here, would you please take a look at this [Flutter-specific thread](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/152#issuecomment-540978949) and try the suggestions from there?

**Comment by dasundev** [comment]
Set `Enable Bitcode` to **NO** It should be work.
![image](https://user-images.githubusercontent.com/54996800/104988113-43af5100-5a3d-11eb-97c5-2b9436a48117.png)

**Comment by dasundev** [comment]
Set **Enable Bitcode** to **NO** Then it should be work.
![image](https://user-images.githubusercontent.com/54996800/104988168-69d4f100-5a3d-11eb-9d0f-11ba61eaa97b.png)

**Comment by xinatcg** [comment]
for Unity, check this thread: https://github.com/OneSignal/OneSignal-Unity-SDK/issues/540#issuecomment-1487246330
